src/main/java/jp/ikedam/jenkins/plugins/extensible_choice_parameter/DatabaseChoiceListProvider.java

```diff
@@ -0,0 +1,419 @@
+            {
+                return FormValidation.error(Messages.DbChoiceListProvider_dbName_empty());
+            }
+                       
+            return FormValidation.ok();
+        }
+        
+        /**
+         * Validate a value inputed for tabledb
+         * 
+         * Checks followings:
+         * * not blank
+         * 
+         * @param tabledb
+         * @return FormValidation object
+         */
+        public FormValidation doCheckTabledb(@QueryParameter String tabledb)
+        {
+            if(StringUtils.isBlank(tabledb))
+            {
+                return FormValidation.error(Messages.DbChoiceListProvider_dbTable_empty());
+            }
+                       
+            return FormValidation.ok();
+        }
+        
+        /**
+         * Validate a value inputed for driverdb
+         * 
+         * Checks followings:
+         * * not blank
+         * 
+         * @param driverdb
+         * @return FormValidation object
+         */
+        public FormValidation doCheckDriverdb(@QueryParameter String driverdb)
+        {
+            if(StringUtils.isBlank(driverdb))
+            {
+                return FormValidation.error(Messages.DbChoiceListProvider_jdbcDriver_empty());
+            }
+                       
+            return FormValidation.ok();
+        }
+        
+    }
+	
+	
+	/**
+	 * Recupere la liste des environnements recuperer de la base de donnees JACK	  
+	 *	 
+	 */
+	/**
+	 * 
+	 * @return
+	 */
+	private List<String> performDbQuery(){
+
+		Connection conn = null;
+		ResultSet resultat = null;
+		List<String> resultList = new ArrayList<String>();
+		try {
+			String driver = getJdbcDriver();
+			if (driver == null || StringUtils.isEmpty(driver)) {
+				LOGGER.log(Level.WARNING,"Invalid driver");
+			}
+			
+			Class.forName(driver);
+			
+			String user = getDbUsername();
+			if (user == null || StringUtils.isEmpty(user)) {
+				LOGGER.log(Level.WARNING,"Invalid user");
+			}
+
+			String password = getDbPassword();
+			if (password == null || StringUtils.isEmpty(password)) {
+				LOGGER.log(Level.WARNING,"Invalid password");
+			}
+
+			String urlBase = getJdbcUrl();
+			if (urlBase == null || StringUtils.isEmpty(urlBase)) {
+				LOGGER.log(Level.WARNING,"Invalid JDBC URL");				
+			}
+			
+			String colomndb = getDbColumn();
+			if (colomndb == null || StringUtils.isEmpty(colomndb)) {
+				LOGGER.log(Level.WARNING,"Invalid column");
+			}
+			
+			String namedb = getDbName();
+			if (namedb == null || StringUtils.isEmpty(namedb)) {
+				LOGGER.log(Level.WARNING,"Invalid DB name");
+			}
+
+			String tabledb = getDbTable();
+			if (tabledb == null || StringUtils.isEmpty(tabledb)) {
+				LOGGER.log(Level.WARNING,"table database invalid");				
+			}
+			
+			conn = DriverManager.getConnection(urlBase, user, password);
+						
+			// By default, a SELECT * is performed
+			String selectQuery = "select " + colomndb + " from " + namedb + "." + tabledb;
```

Comment:
This can be a vulnerability.
for example, `columndb = "1; DROP TABLE configuration; SELECT somefield"`